### DISCUSSION TITLE
nextjs 14 pass pass runtime environment configuration values while using docker

### DISCUSSION THREAD

**Discussion by tomani77** [vercel/next.js#60465] [Help]
### Summary

I am building an app using Next.JS 14. I have configuration properties (to be read on server side) defined in .env file. My requirement is to externalize the property i.e. I should be able to pass it during container start so that same image could be used in multiple environments. Could you let me know how to achieve it.

I have tried following which does not work

.env file

`API_HOSTNAME=http://VariesFromEnvironmentToEnvironment`

This is how I am reading in Code in component

```
process.env.API_HOSTNAME
process.env["API_HOSTNAME"]
```
Docker File

```
FROM node:18

WORKDIR /app
COPY package*.json ./
# Install dependencies  
RUN npm ci  

# Copy the app source code to the container  
COPY . .  

# Build the Next.js app  
RUN npm run build  

# Expose the port the app will run on  
EXPOSE 3000  

# Start the app  
CMD ["npm", "start"]
```
Docker Build - `docker build -t poc-nextjs-env  . Run Container- docker run --env API_HOSTNAME=updatedvalue  -p 3000:3000 poc-nextjs-env`

expected value of `process.env.API_HOSTNAME - updatedvalue`

Actual value of `process.env.API_HOSTNAME - http://VariesFromEnvironmentToEnvironment (same as mentioned in .env file)`

**Comment by tomani77** [comment]
Could someone suggest?

**Reply by ezeparziale** [reply]
Attach the shell with 

`docker exec -it YOUR_ID_IMAGE bash`

And check if you copy your env file inside the docker image.
